src/snowflake/snowpark/dataframe.py

```diff
@@ -916,23 +916,36 @@ def to_local_iterator(
             **kwargs,
         )
 
-    def __copy__(self) -> "DataFrame":
-        """Implements shallow copy protocol for copy.copy(...)."""
+    def _copy_plan(self) -> LogicalPlan:
+        """Returns a shallow copy of the plan of the DataFrame."""
         if self._select_statement:
             new_plan = copy.copy(self._select_statement)
             new_plan.column_states = self._select_statement.column_states
             new_plan._projection_in_str = self._select_statement.projection_in_str
             new_plan._schema_query = self._select_statement.schema_query
             new_plan._query_params = self._select_statement.query_params
+            return new_plan
         else:
-            new_plan = copy.copy(self._plan)
+            return copy.copy(self._plan)
 
-        # TODO SNOW-1762416: Clarify copy-behavior in AST. For now, done as weak-copy always. Yet, we may want to consider
-        # a separate AST entity to model deep-copying. A deep-copy would generate here a new ID different from self._ast_id.
-        # We additionally need to consider behavior when a user calls copy.copy() on a Snowpark Dataframe. For now, consider
-        # all calls internal, and leave AST handling to the caller.
-        df = DataFrame(self._session, new_plan, _emit_ast=False)
-        return df
+    def _copy_without_ast(self) -> "DataFrame":
+        """Returns a shallow copy of the DataFrame with the AST enabled."""
```

Comment:
```suggestion
    def _copy_without_ast(self) -> "DataFrame":
        """Returns a shallow copy of the DataFrame without the AST enabled."""
```
should we add without here?